storage/database/db_manager_test.go

```diff
@@ -341,6 +353,154 @@ func TestDBManager_IstanbulSnapshot(t *testing.T) {
 	}
 }
 
+// TestDBManager_TrieNode tests read and write operations of state trie nodes.
+func TestDBManager_TrieNode(t *testing.T) {
+	for _, dbm := range dbManagers {
+		cachedNode, _ := dbm.ReadCachedTrieNode(hash1)
+		assert.Nil(t, cachedNode)
+		hasStateTrieNode, _ := dbm.HasStateTrieNode(hash1[:])
+		assert.False(t, hasStateTrieNode)
+
+		batch := dbm.NewBatch(StateTrieDB)
+		if err := batch.Put(hash1[:], hash2[:]); err != nil {
+			t.Fatal("Failed putting a row into the batch", "err", err)
+		}
+		if _, err := WriteBatches(batch); err != nil {
+			t.Fatal("Failed writing batch", "err", err)
+		}
+
+		cachedNode, _ = dbm.ReadCachedTrieNode(hash1)
+		assert.Equal(t, hash2[:], cachedNode)
+
+		if err := batch.Put(hash1[:], hash1[:]); err != nil {
+			t.Fatal("Failed putting a row into the batch", "err", err)
+		}
+		if _, err := WriteBatches(batch); err != nil {
+			t.Fatal("Failed writing batch", "err", err)
+		}
+
+		cachedNode, _ = dbm.ReadCachedTrieNode(hash1)
+		assert.Equal(t, hash1[:], cachedNode)
+
+		stateTrieNode, _ := dbm.ReadStateTrieNode(hash1[:])
+		assert.Equal(t, hash1[:], stateTrieNode)
+
+		hasStateTrieNode, _ = dbm.HasStateTrieNode(hash1[:])
+		assert.True(t, hasStateTrieNode)
+	}
+}
+
+// TestDBManager_TxLookupEntry read, write and delete operations of TxLookupEntries.
+func TestDBManager_TxLookupEntry(t *testing.T) {
+	tx, err := genTransaction(num1)
+	if err != nil {
```

Comment:
How about using `assert.NoError()`?
```suggestion
	assert.NoError(t, err, "Failed to generate a transaction")
```